### PR TITLE
Issue9 property or default

### PR DESCRIPTION
**Pull request by SimonNodel-AI**

### CODE REVIEW

**Comment by SimonNodel-AI** [comment]
@JesseChezenko-AI do you mind having a quick look.

File: README.md [hunk 1]
```diff
@@ -47,10 +47,14 @@ Returns true if value is a string of non zero length.
...
+Returns the value of the property if it not null or undefined otherwise, it will return a default value unless, defaultValue is a function in which case, it will return the result of calling default value function with remainder of arguments.
+Note: propertyPath format is 'dot.delimited.path'.
 
 ### stringOrDefault(value, defaultValue, ...)
-Returns value if value is a non empty string otherwise, if defaultValue is a function it will return the result of calling it with remainder of arguments or defaultValue.
+Returns value if value is a non empty string otherwise, it will return a default value unless, defaultValue is a function in which case, it will return the result of calling default value function with remainder of arguments.
 
 ### valueOrDefault(value, defaultValue, ...)
-Return value if value is not null or undefined otherwise, if defaultValue is a function it will return the result of calling it with remainder of arguments or defaultValue.
+Return value if value is not null or undefined otherwise, it will return a default value unless, defaultValue is a function in which case, it will return the result of calling default value function with remainder of arguments.
```
Comment (JesseChezenko-AI) [inline README.md:60]: PR looks good but I'm not sure I like the API design of taking the rest arguments and calling the default function with them..

do you have a use case in mind?

Comment (SimonNodel-AI) [inline README.md:60]: Sure:

```
propertyOrDefault(someObject, 'some.path.to.property', doSomethingSpecial, someOtherAdditionalObjectToMakeThingsSpecial);
```

Comment (SimonNodel-AI) [inline README.md:60]: It makes less sense for inline functions but, useful for functions that are not inlined.

Comment (JesseChezenko-AI) [inline README.md:60]: The function could have those arguments available via a closure or you could compute the value before you call propertyOfDefault.

Closure Example

```
   var someOtherAdditionalObjectToMakeThingsSpecial = ...;
    function doSomethingSpecial(){
          ...
          ...someOtherAdditionalObjectToMakeThingsSpecial...
          ...
    }
    propertyOrDefault(someObject, 'some.path.to.property', doSomethingSpecial);

```

Computed example

```
    resultOfSomethingSpecial =   ...someOtherAdditionalObjectToMakeThingsSpecial... + ...;
    propertyOrDefault(someObject, 'some.path.to.property', resultOfSomethingSpecial);

```

Comment (JesseChezenko-AI) [inline README.md:60]: I guess if you don't have control of the method signature then it makes more sense...

even then though you could wrap the function in an anonymous function and provide the arguments that way

Comment (JesseChezenko-AI) [inline README.md:60]: anyways... something to think about.. I don't feel that strongly about it.

Comment (SimonNodel-AI) [inline README.md:60]: I pondered more on this and I agree, this should be simpler. I'll remove the handling of additional arguments.

**Pull request merged**